paramz/core/parentable.py

```diff
@@ -73,3 +75,26 @@ def _notify_parent_change(self):
         Dont do anything if in leaf node
         """
         pass
+
+    def __getstate__(self):
+        dc =  super().__getstate__()
+        dc.pop('_parent_', None)
+        return dc
+    
+    def __deepcopy__(self, memo: dict):
+        s = self.__new__(self.__class__) # fresh instance
+        memo[id(self)] = s # be sure to break all cycles --> self is already done
+        # The above line can cause hard to understand exceptions/bugs, because s is not 'done' its state attributes need to be copied first
+        # If a state attribute has a link to a parent object, the parent is copied first, using the uninitialized copy s
+        # thereby throwing an exception when attributes of the child are accessed while copying the parent.
+        # so a subclass should not link to its parent or handel the link like its done here.
```

Comment:
I am not quite sure whether this is "documentation", a "hint" or something that should be adressed before we merge. Could you comment on that?

Reply:
Unfortunately it seems like some of my comments have disappeared (doing reviews on the train might not be a good idea...)

The comments that disappeared from the review are mainly about the TODO's that are within your changes. I was wondering whether you consider them TODOs within your PR or more like a general TODO that is to be adressed outside of this PR?

Reply:
Hello,

They are general TODO that are to be addressed outside of this PR.
The PR fixes the according bug, and no additional changes are required.

However, while fixing the bug, I stumbled over a few other "design decisions" in the old code that could/should need refactoring and can potentially cause additional hard-to-find bugs.
So, I wrote some TODOs and some explanatory comments so that I or somebody else can find them again faster and fix them (maybe with another PR).

A working test has already been provided in my initial comment.
If no exception is thrown, everything is fine ;) -> If a similar bug appears, it will likely result in an exception with the provided test.

To thoroughly test for similar bugs, every module that supports copy and/or deep copy should be copied and checked to see if the copy is identical.
(But I think this is a long time job, and nothing I can provide)

I hope i could help